packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/SubjectGroupViewer/SubjectGroupViewerContainer.js

```diff
@@ -39,15 +45,41 @@ function storeMapper (stores) {
   const cellStyle = viewerConfig.cell_style || DEFAULT_CELL_STYLE
   const gridColumns = viewerConfig.grid_columns || DEFAULT_GRID_COLUMNS
   const gridRows = viewerConfig.grid_rows || DEFAULT_GRID_ROWS
+  
+  const {
+    activeStepTasks
+  } = stores.classifierStore.workflowSteps
+  const [currentTask] = activeStepTasks.filter(task => task.type === 'subjectGroupComparison')
+  
+  const {
+    addAnnotation,
+    active: classification,
+  } = stores.classifierStore.classifications
+  
+  const isCurrentTaskValidForAnnotation = !!currentTask
+
+  // Note: the Task's Annotations are initialised by the SubjectGroupComparisonTask
+  // component. However, do note that it's possible to have a
+  // SubjectGroupViewer without a SubjectGroupComparisonTask.
+  
+  // Currently only supports 1 Subject Group Task in the workflow.
+  const allAnnotations = classification ? Array.from(classification.annotations.values()) : []
+  const annotation = allAnnotations.find(item => (getType(item).name === 'SubjectGroupComparisonAnnotation'))
```

Comment:
```suggestion
  // Currently only supports 1 Subject Group Task in the workflow.
  const annotation = classification?.addAnnotation(currentTask)
```
This might be a safer way to get the annotation for the current task, without throwing React warnings.

Reply:
`classification?.annotation(currentTask)` might work too. I think there's an `annotation(task)` view that gives you back the annotation for a specific task (or possibly a specific task key.)

Reply:
Unfortunately, `classification?.annotation(currentTask)` doesn't work either. 😞 See comment: https://github.com/zooniverse/front-end-monorepo/pull/1638#issuecomment-642219935 (sorry, it's kinda buried under so many revisions!)

I'll experiment with `const annotation = classification?.addAnnotation(currentTask)`, thank you for the suggestion. Two things for me to note:
- At the moment, the SubjectGroupViewer assumes that if there's a SGComparisonTask, it'd be initialised by the Task component. (`TaskArea/components/Tasks/Tasks.js`) Gotta check there's no issue with double-initialisation, if that's such a thing.
- Check that the `addAnnotation` doesn't trigger if there's a non-SGComparisonTask on the right-hand panel. (e.g. a Single Choice Task with an SGViewer.) (Shouldn't be the case if I remember to set currentTask = undefined for non-SubjectGroupXTasks)

Reply:
Oh wait, you want to continue to display the annotation during subsequent tasks, don't you? My suggestion would only display it while we're doing the subject selection task. It would vanish on moving to the next task in the workflow. I missed that because I only looked at one task in the test workflow, so the current task was always a subject group task.